src/flow.ts

```diff
@@ -806,6 +830,20 @@ export class Flow {
         thisLocalFlags[i] = newFlags;
       }
     }
+
+    // Only the most right flow will have an effect
+    // on the resulting flow.
```

Comment:
This seems to miss

```ts
if (cond) {
  // do nothing in left flow
} else {
  this.theField = init; // initialize in right flow
}
```

Reply:
Right, I hadn't correctly verified this, but added a test case for this one: https://github.com/AssemblyScript/assemblyscript/pull/1155/files#diff-5fa1688a93849f802b2ae2b7f3cbffe6R26, it's working as expected.

Also I _think_ it's very similar to: https://github.com/AssemblyScript/assemblyscript/pull/1155/files#diff-5fa1688a93849f802b2ae2b7f3cbffe6R48

Reply:
Hmm, the new test should produce an error, but it doesn't seem to do so. Unless I'm missing something :)

Reply:
I misinterpreted your message, yes, indeed. The field should only be initialiazed in the resulting flow if both flows have initialized it.